watsononlinestore/watson_online_store.py

```diff
@@ -242,13 +243,100 @@ def get_watson_response(self, message):
         return response
 
     @staticmethod
-    def format_discovery_response(response):
-        """Specific to ibm_store_html data
-        """
+    def format_discovery_response(response, data_source):
+        """Format results based on discovery data source."""
         output = []
         if not ('results' in response and response['results']):
             return output
 
+        def get_product_name(entry, data_source):
+            """ Pull product name from entry data for nice user display.
+
+            param: string data_source The name of the discovery data source
+            """
+            product_name = ""
+
+            if data_source == "amazon":
+                # Pull out product name from enriched metadata.
+                if 'extracted_metadata' in entry:
+                    metadata = entry['extracted_metadata']
+                    if 'title' in metadata:
+                        product_name = metadata['title']
+            else:
+                # Pull out product name from page text.
+                if 'text' in entry:
+                    product_tag = "Product:"
+                    category_tag = "Category:"
+                    text = entry['text']
+                    sidx = text.find(product_tag)
+                    if sidx > 0:
+                        sidx += len(product_tag)
+                        eidx = text.find(category_tag, sidx, len(text))
+                        if eidx > 0:
+                            product_name = text[sidx:eidx-1]
+
+            return product_name
+
+        def get_product_url(entry, data_source):
+            """ Pull product url from entry data so user can navigate
+            to product page.
+
+            param: string data_source The name of the discovery data source
+            """
+            product_url = ""
+
+            if 'html' in entry:
+                html = entry['html']
+
+                if data_source == "amazon":
+                    # URL is found at the end of the html doc.
+                    href_tag = "<a href="
+                    sidx = html.rfind(href_tag)
+                    if sidx > 0:
+                        sidx += len(href_tag)
+                        eidx = html.find('>', sidx, len(html))
+                        if eidx > 0:
+                            product_url = html[sidx+1:eidx-1]
+                else:
```

Comment:
We should probably call out that this is for ibm_store, ie.:
elif data_source == "ibm_store"